observable.go

```diff
@@ -234,20 +235,28 @@ func (o *observable) Take(nth uint) Observable {
 // a new Observable with the taken items.
 func (o *observable) TakeLast(nth uint) Observable {
 	f := func(out chan interface{}) {
-		buf := make([]interface{}, nth)
+		n := int(nth)
+		r := ring.New(n)
+		count := 0
 		it := o.iterable.Iterator()
 		for {
 			if item, err := it.Next(); err == nil {
-				if len(buf) >= int(nth) {
-					buf = buf[1:]
-				}
-				buf = append(buf, item)
+				count++
+				r.Value = item
+				r = r.Next()
 			} else {
 				break
 			}
 		}
-		for _, takenItem := range buf {
-			out <- takenItem
+		if count < n {
+			for i := 0; i < n-count; i++ {
```

Comment:
Could be replaced by a standard call:
```go
if count < n {
	r = r.Move(n - count)
	n = count
}
```

Or, maybe even better:
```go
if count < n {
	remaining := n - count
	if remaining <= count {
		r = r.Move(n - count)
	} else {
		r = r.Move(-count)
	}
	n = count
}
```

In the case where we have only one element in the ring buffer but the initial size was 1 million for example, it's better to move backward.

If you agree with this, please just add a test to cover `if remaining <= count` and another one for the `else`.